### ISSUE TITLE
A rewording of 1.4.12 Text spacing that makes it applicable to a broader range of technologies

### ISSUE THREAD

**Issue by mapluke** [w3c/wcag21#656]
The following is a consensus position amongst those participating in the work of the European Team developing a revision of EN 301 549 which will be aligned with WCAG 2.1:

In the current SC wording, the phrase “If the technologies being used allow the user agent to set” may not always make sense if the SC is applied to software and document technologies like pdf.

The current wording “setting all of the following“ leaves open the question of what should happen if only a subset of these are settable?

The following formulation, before the original bullets, appears to solve both of those issues:

@@If a mechanism is available to set the following@@ text style properties, then no loss of content or functionality occurs by setting all of the following @@ that are supported, while @@ changing no other style property:

Adding the following exception also makes this SC applicable to applications like PowerPoint that have fixed size pages that act as a canvas:

Except when a fixed size content layout area is essential to the information being conveyed.

**Comment by alastc** [comment]
Hi @mapluke,

(Not official response, follow up questions.)

> In the current SC wording, the phrase “If the technologies being used allow the user agent to set” may not always make sense if the SC is applied to software and document technologies like pdf.

PDF allows that over-riding, but powerpoint doesn't unless you change it as an author.

On the exception: Do any applications that allow the user to set text-styles also use that canvas approach? For the reflow SC we included a similar exception but scoped it to things which had to show mutiple things at the same time (e.g. content canvas and toolbars), so something intrinsic to the content.

I'm concerned that adding that exception means that any page with a fixed with in then out-of-scope, but web content (HMTL, PDF, ePub) in general does allow you to adjust layout. Not that you would have to, leaving buffer around text is the core technique, which should be possible from any authoring environment.

There was a long discusion about 'mechanism' language, and virtually everyone (who isn't a WCAG-wonk) reads it as having to add an on-page widget, which we really want to avoid.

The other option considered was to scope it to markup languages, something like:
> In content implemented using markup languages, then no loss of content or functionality occurs by setting all of the following style properties, while changing no other style property:

Would that prevent/avoid the concern?

**Comment by mapluke** [comment]
I think that:

- In content implemented using markup languages, then no loss of content or functionality occurs by setting all of the following style properties, while changing no other style property:

Would prevent/avoid the concern. I wonder though whether it still needs the "that are supported," to cater for markup languages that might not support all of the 4 bulleted items. This would then read:

- In content implemented using markup languages, then no loss of content or functionality occurs by setting all of the following style properties that are supported, while changing no other style property:

**Comment by kerstinp** [comment]
Hi and sorry for jumping in from the outside (and as non-member of the Working Group)

@alastc 

> PDF allows that over-riding

In which PDF-Reader are these over-ridings possible? And I mean a _real_ PDF-Reader. Not a software like VIP-Reader which displays the _converted_ content of a PDF.

**Comment by alastc** [comment]
That is an extreme version of over-riding.

However, if we go down the route of "In content implemented using markup languages" then it doesn't apply to PDF.

**Comment by kerstinp** [comment]
Hi @alastc 

I would say that it is no over-riding at all, cause the conversion process is before displaying the content in this software. And yes: When the WG uses "content implemented using markup languages" then it doesn't apply to PDF.

But the question is still open: In which PDF-Reader is the over-riding mentioned in the SC possible?

(I'm not a member of the WG but I would prefer the suggested wording posted by @mapluke )

**Comment by alastc** [comment]
Hi @kerstinp,

From a user-piont of view, it is easy to see VPI-reader as overriding, in the same way that Stylish does in a browser. I'm very aware of your previous comments about the feasibility of that in PDFs. I don't think the group wanted to prevent this SC from progressing due to the issues with user-agents for PDFs. However, we do avoid those issues by moving to a markup-only approach.

On Mike's phrasing, we still have the issue with "if a mechanism is available" leading to people thinking you need to add an on-page widget, which is why we wanted to avoid that phrasing.

**Comment by kerstinp** [comment]
Hi @alastc, this is sorry to say still no answer.

But I don't want to capture this item and repeat this question in my comments. 

Sorry @mapluke for jumping in. I don't want to capture your item.

**Comment by mapluke** [comment]
@alastc , 

I certainly cannot think of markup languages that don't allow all of the things in the bulleted list. So I am happy with your original suggestion

>In content implemented using markup languages, then no loss of content or functionality occurs by setting all of the following style properties, while changing no other style property:

Unless I'm missing something, the debate over whether or not changing the properties is supported for PDF (not I believe) is not relevant when considering this markup language-based wording.

I'm still not really familiar with how and when this issue will get resolved (or not). Can someone inform me. 

Having wording like this would be very valuable in Europe in deciding whether this SC could be directly applied to software or document technologies without having to do WCAG2ICT type rewording to suit these contexts.

**Comment by alastc** [comment]
Hi @kerstinp,

The answer is that VIP-reader does allow for it, with notable restrictions on the content that would work. Whether it re-renders the content or applies it with styles isn't relevent to the SC text, even though it has implications for authors. Is that a satisfactory answer?

The only real options are to: 

1. keep the scope wide to include things which makes life very difficult for some authors (for PDF, and app-focused contexts), or 
2. narrow the scope to where we know it is feasible.

Given the state of PDF readers, and that we can (and should) have this type of SC for web-content, what would you suggest?

@mapluke I think if that wording is ok then @lauracarlson (or @awkawk?) would update it and put forward for a CFC. I think. It is well within the deadline, so should be fine.

**Comment by lauracarlson** [comment]
Hi @alastc , 

I believe @awkawk assigned himself this issue and is working on a [proposed response for AGWG consideration](https://www.w3.org/WAI/GL/wiki/Draft_Responses_to_Dec_WD_Issues#656). I suspect a survey question, discussion, and CFC will likely follow that. 

@allanj-uaag , @slhenry ,  you may be interested in the proposed scope narrowing.

**Comment by alastc** [comment]
Ok, I've updated the proposed response on the wiki to match, thanks.

**Comment by kerstinp** [comment]
@alastc Let's discuss the PDF-Reader-thing [here](https://github.com/w3c/wcag21/issues/644)

**Comment by lauracarlson** [comment]
Hi @alastc ,

If I recall correctly, I "think" the Chairs wanted to limit long discussions on GitHub in lieu of the following process: 

1. Proposal on the Wiki
2. Survey ([This one is in the Resolving issues #2 survey](https://www.w3.org/2002/09/wbs/35422/resolving_issues_2/results#xq13))
3. Telco discussion
4. CFC 

@awkawk  @joshueoconnor is this the preferred process? Or is it okay to continue the conversation here?

**Comment by awkawk** [comment]
Yes, that is the preferred process...

**Comment by awkawk** [comment]
(Official WG Response) 
Thank you for your comment, we have integrated a change to avoid issues with PDF and application scenarios, so the SC text starts:

"In content implemented using markup languages that support the following text style properties, no loss of content or functionality occurs by setting all of the following and by changing no other style property."

The purpose of the SC is to encourage people to enable over-riding of content via user-agent mechanisms, rather than require on-page widgets. Therefore the working group would like to avoid 'mechanism is available' language for this SC as people tend to read it as meaning adding a widget to a page.
